### PR TITLE
Feature/image detail

### PR DESCRIPTION
**Pull request by deven98**
Implemented Image detail screen

### CODE REVIEW

File: lib/src/image_actions_modal.dart [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        title,
+        style: color == null ? titleStyle : titleStyle.copyWith(color: color),
+      ),
+      leading: Icon(
+        iconData,
+        color: color ?? StreamChatTheme.of(context).primaryIconTheme.color,
+        size: 24.0,
+      ),
+      onTap: onTap,
+      tileColor: Colors.white,
```
Comment (xsahil03x) [inline lib/src/image_actions_modal.dart:167]: Splash is not visible due to the white tile color, Maybe remove this line.

**Review by xsahil03x — CHANGES_REQUESTED** [review]
I have mentioned some of my findings, Rest LGTM.

File: lib/src/image_actions_modal.dart [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                            StreamIcons.curve_line_down_left, () {}),
+                        _buildButton(context, 'Show in Chat', StreamIcons.eye,
+                            () {
+                          Navigator.pop(context);
+                          Navigator.pop(context);
+                        }),
+                        _buildButton(context, 'Save Image', StreamIcons.save_1,
+                            () async {
+                          await GallerySaver.saveImage(
+                              urls[currentIndex].split('?')[0]);
```
[the remark this answers is not in the document]
Comment (imtoori) [reply lib/src/image_actions_modal.dart:122]: it doesn't fail on ios, but it's slow
we could add a loading indicator

**Comment by imtoori** [comment]
@deven98  I think we should handle videos as well

File: lib/src/image_actions_modal.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (xsahil03x) [reply lib/src/image_actions_modal.dart:122]: Looks like it was some emulator bug, but there is still one error. Most of the image paths don't have a valid image extension.

eg : https://picsum.photos/seed/53/200/200

so this throws an exception

```dart[ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: Invalid argument(s): File on path is not an image.
E/flutter (28315): #0      GallerySaver.saveImage (package:gallery_saver/gallery_saver.dart:50:7)
E/flutter (28315): #1      ImageActionsModal._buildPage.<anonymous closure> (package:stream_chat_flutter/src/image_actions_modal.dart:121:46)
E/flutter (28315): #2      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:993:19)
E/flutter (28315): #3      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:1111:38)
E/flutter (28315): #4      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:183:24)
E/flutter (28315): #5      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:598:11)
E/flutter (28315): #6      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:287:5)
E/flutter (28315): #7      BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:259:7)
E/flutter (28315): #8      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:157:27)
E/flutter (28315): #9      GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:362:20)
E/flutter (28315): #10     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:338:22)
E/flutter (28315): #11     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:267:11)
E/flutter (28315): #12     GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:295:7)
E/flutter (28315): #13     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:240:7)
E/flutter (28315): #14     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:213:7)
E/flutter (28315): #15     _rootRunUnary (dart:async/zone.dart:1206:13)
E/flutter (28315): #16     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
E/flutter (28315): #17     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1005:7)
E/flutter (28315): #18     _invoke1 (dart:ui/hooks.dart:265:10)
E/flutter (28315): #19     _dispatchPointerDataPacket (dart:ui/hooks.dart:174:5)
```

[the remark this answers is not in the document]
Comment (deven98) [reply lib/src/image_actions_modal.dart:122]: This is going to be problematic, I'll see what I can do

**Pull request merged**